src/profile.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include "iobuf.h"
+
+#if MG_ENABLE_PROFILE
+struct mg_profitem {
+  const char *name;
+  uint32_t timestamp;
+};
+
+#define MG_PROF_ADD(c, name_)                                          \
+  do {                                                                 \
+    struct mg_iobuf *io = &c->prof;                                    \
+    struct mg_profitem *first = (struct mg_profitem *) io->buf;        \
+    struct mg_profitem item = {name_, (uint32_t) mg_millis()};         \
+    if (io->len >= sizeof(*first)) item.timestamp -= first->timestamp; \
+    mg_iobuf_add(io, io->len, &item, sizeof(item));                    \
+  } while (0)
+
+#define MG_PROF_INIT(c)                \
+  do {                                 \
+    mg_iobuf_init(&(c)->prof, 0, 256); \
+    MG_PROF_ADD((c), "init");          \
+  } while (0)
+
+#define MG_PROF_FREE(c) mg_iobuf_free(&(c)->prof)
+
+#define MG_PROF_DUMP(c)                                               \
+  do {                                                                \
+    struct mg_iobuf *io = &c->prof;                                   \
+    struct mg_profitem *p = (struct mg_profitem *) io->buf;           \
+    struct mg_profitem *e = &p[io->len / sizeof(*p)];                 \
+    MG_INFO(("%lu profile:", c->id));                                 \
+    while (p < e) {                                                   \
+      MG_INFO(("%5lx %s", (unsigned long) p->timestamp, p->name)); \
+      p++;                                                            \
```

Comment:
These will most likely have the same timestamp, we could be less intrusive by combining everything in one log. Or would alloc+free be worse ? I think printing 5 extra timestamps for 5 events is quite more time consuming.

Reply:
Printing time is not a worry at all.
Added comments!

Reply:
Not here, yes, we are not queing. Maybe for huge file serving it may be a concern